OpenSSL-Strategy.md

```diff
@@ -0,0 +1,361 @@
+  those packages.  Therefore, the floating patches do not apply and it is
+  possible that a different version of OpenSSL 1.0.2 is in use by Node.js than
+  the version that was shipped in the source tree.
+* **NodeSource Builds**: Binaries
+  [distributed](https://github.com/nodesource/distributions) by NodeSource are
+  heavily used, and do not follow the Linux distribution policyof dependency
+  seperation. NodeSource uses the Node.js defaults for `configure` and compile
+  and statically link in the OpenSSL source shipped bundled with Node.js.
+* **Homebrew**: The
+  [formula](https://github.com/Homebrew/homebrew-core/blob/master/Formula/node.rb)
+  for Node.js as distributed on macOS by the popular `brew` command, by default,
+  will compile using the default `configure` configuration. It is possible to
+  override this using a `with-openssl` option which will compile against the
+  version of OpenSSL that was most recently installed with `brew` but this is
+  not believed to be in common use.
+
+
+## Node.js version-specific strategy
+
+### Node.js version 4.x (EOL 2018-04-30)
+
+No longer maintained.  Not discussed further.
+
+### Node.js versions 6.x (EOL 2019-04-30) and 8.x (EOL 2019-12-31)
+
+Node.js 6.x and 8.x have the same OpenSSL versions and policies.
+
+Node.js 8.x is unusual because it's end-of-life would normally be April, 2020,
+but it was moved earlier to conincide with the end-of-life of OpenSSL 1.0.2.
+
+* OpenSSL version: 1.0.2 (kept up-to-date with latest)
+* Allowed shared OpenSSL version: 1.0.2x
+* Default minimum TLS version is TLSv1, default maximum is TLSv1.2. No other
+  TLS versions are supported by this version of OpenSSL or Node.js.
+* FIPS: supports being configured to build against OpenSSL FIPS 2.0. FIPS
+  configuration is not default, and OpenSSL FIPS 2.0 is not included in
+  Node.js.
+
+XXX Do we support FIPS builds, or do we treat like other shared builds, best
+effort to fix but not release critical if its broken?
+
+
+### Node.js version 10.x (EOL April-2021)
+
+OpenSSL LTS support timing, the lack of OpenSSL LTS planning and the lack of a
+clear timeframe for a new FIPS module complicates Node.js 10.
+
+For Node.js < 10.16.0:
+* OpenSSL version: 1.1.0
+* Allowed shared OpenSSL version: 1.0.2 or 1.1.0
+* Default minimum TLS version is TLSv1, default maximum is TLSv1.2. No other
+  TLS versions are supported by this version of OpenSSL or Node.js.
+* FIPS: not supported
+
+For Node.js >= 10.16.0:
+* OpenSSL version: 1.1.1
+* Allowed shared OpenSSL version: 1.0.2 or 1.1.0 or 1.1.1
+* Default minimum TLS version is TLSv1, default maximum is TLSv1.2. It is
+  expected that TLS1.3 support will be backported to 10.x, but it will *not* be
+  supported by default, only by explicit run-time configuration.
+* FIPS: not supported
+
+XXX Check whether our test matrix is updated to test this.
+XXX Node.js 10.x allows building against an external OpenSSL 1.0.2, so doesn't
```

Comment:
I haven't tried (yet) but the `./configure` option of 10.x sure makes it look like it supports building against FIPS.

Reply:
It does not, because it won't build against 1.0.2